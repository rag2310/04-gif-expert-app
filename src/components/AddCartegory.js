import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCartegory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(category => [inputValue, ...category]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCartegory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

