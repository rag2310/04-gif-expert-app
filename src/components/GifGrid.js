import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data: imagenes, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__bounce">{category}</h3>

            {loading && <p className="animate__animated animate__flash">Loading</p>}

            <div className="card-grip">
                {
                    imagenes.map((img) =>
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />)
                }
            </div>
        </>
    );
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}