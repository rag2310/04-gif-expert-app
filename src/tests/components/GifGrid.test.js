import {shallow} from "enzyme";
import '@testing-library/jest-dom';
import {GifGrid} from "../../components/GifGrid";
import {useFetchGifs} from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas de <GifGrid />', function () {

    const category = 'One Punch';

    test('debe mostrar <GifGrid />', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetch()', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost:8080.jpg',
            title: 'Cualquir cosa'
        }, {
            id: '123',
            url: 'https://localhost:8080.jpg',
            title: 'Cualquir cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});