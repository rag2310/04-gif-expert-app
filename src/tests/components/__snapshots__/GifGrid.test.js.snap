// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetch() 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    One Punch
  </h3>
  <div
    className="card-grip"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquir cosa"
      url="https://localhost:8080.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquir cosa"
      url="https://localhost:8080.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas de <GifGrid /> debe mostrar <GifGrid /> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grip"
  />
</Fragment>
`;
